applications/asset_tracker_v2/tests/ui_module/src/ui_module_test.c

```diff
@@ -0,0 +1,401 @@
+	TEST_ASSERT_EQUAL(0, UI_MODULE_EVT_HANDLER((struct event_header *)_event))
+
+/* Dummy structs to please linker. The EVENT_SUBSCRIBE macros in ui_module.c
+ * depend on these to exist. But since we are unit testing, we dont need
+ * these subscriptions and hence these structs can remain uninitialized.
+ */
+struct event_type __event_type_gnss_module_event;
+struct event_type __event_type_debug_module_event;
+struct event_type __event_type_app_module_event;
+struct event_type __event_type_data_module_event;
+struct event_type __event_type_cloud_module_event;
+struct event_type __event_type_modem_module_event;
+struct event_type __event_type_sensor_module_event;
+struct event_type __event_type_ui_module_event;
+struct event_type __event_type_util_module_event;
+
+/* The following is required because unity is using a different main signature
+ * (returns int) and zephyr expects main to not return value.
+ */
+extern int unity_main(void);
+
+/* Suite teardown finalizes with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+void setUp(void)
+{
+	mock_modules_common_Init();
+	mock_event_manager_Init();
+
+	/* Reset internal module states. */
+	state = 0;
+	sub_state = 0;
+	sub_sub_state = 0;
+
+	/* Clear internal LED pattern transition list. */
+	transition_list_clear();
+}
+
+void tearDown(void)
+{
+	mock_modules_common_Verify();
+	mock_event_manager_Verify();
+}
+
+/* Handler that validates events sent from the UI module. */
+static void validate_ui_evt(struct event_header *eh, int no_of_calls)
+{
+	struct ui_module_event *event = cast_ui_module_event(eh);
+
+	TEST_ASSERT_EQUAL(UI_EVT_SHUTDOWN_READY, event->type);
+}
+
+/* Function used to verify the internal state of the UI module. */
+static void state_verify(enum state_type state_new, enum sub_state_type sub_state_new,
+			 enum sub_sub_state_type sub_sub_state_new)
+{
+	TEST_ASSERT_EQUAL(state_new, state);
+	TEST_ASSERT_EQUAL(sub_state_new, sub_state);
+	TEST_ASSERT_EQUAL(sub_sub_state_new, sub_sub_state);
+}
+
+/**
+ * @brief Function that validates the next node in the led pattern_transition_list.
+ *	  If the passed in led_state and duration_sec variables are 0 the function returns
+ *	  true if no more nodes are present in the list.
+ *
+ * @param led_state LED state that will be verified against.
+ * @param duration_sec Duration of the LED state.
+ *
+ * @return true if the LED state and duration is associated with the next node.
+ * @return false if the LED state and duration is not associated with the next node.
+ *
+ */
+static bool list_node_verify_next(enum led_state led_state, int16_t duration_sec)
+{
+	struct led_pattern *led_pattern = NULL;
+	sys_snode_t *node = sys_slist_get(&pattern_transition_list);
+
+	if (node == NULL && led_state == 0 && duration_sec == 0) {
```

Comment:
For readability, I'd prefer parentheses around comparisons, it's just more obvious at a glance what's being checked

Reply:
Agreed, will add.